tensorflow_probability/python/experimental/mcmc/particle_filter.py

```diff
@@ -354,130 +354,125 @@ def particle_filter(observations,
       static_trace_allocation_size = 0
       trace_criterion_fn = never_trace
 
-    initial_weighted_particles = _particle_filter_initial_weighted_particles(
-        observations=observations,
-        observation_fn=observation_fn,
-        initial_state_prior=initial_state_prior,
-        initial_state_proposal=initial_state_proposal,
-        num_particles=num_particles,
-        seed=init_seed)
-    propose_and_update_log_weights_fn = (
-        _particle_filter_propose_and_update_log_weights_fn(
-            observations=observations,
-            transition_fn=transition_fn,
-            proposal_fn=proposal_fn,
-            observation_fn=observation_fn,
-            num_transitions_per_observation=num_transitions_per_observation))
-
-    kernel = smc_kernel.SequentialMonteCarlo(
+    if initial_state_proposal is None:
+        initial_state_proposal = initial_state_prior
+
+    # Initializing particles and weights
+    initial_particles, initial_log_weights = \
+        initial_state_proposal.experimental_sample_and_log_prob(
+            [num_particles], seed=seed
+        )
```

Comment:
It would be faster and more stable to avoid computing the log_prob at all when `initial_state_proposal` is None.